sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,192 @@
+from sunpy.visualization.imageanimator import ImageAnimatorWCS
+import matplotlib.pyplot as plt
+import sunpy.visualization.wcsaxes_compat as wcsaxes_compat
+import astropy.units as u
+import sunpycube.wcs_util
+import astropy.nddata
+import numpy as np
+
+__all__ = ['NDCube', 'Cube2D', 'Cube1D']
+
+
+class NDCube(astropy.nddata.NDData):
+    """docstring for NDCube"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, **kwargs):
+        if data.ndim is not wcs.naxis:
+            _naxis = wcs._naxis
+            count = 0
+            for i in _naxis:
+                if not i is 1:
+                    count += 1
+            if count is not data.ndim:
+                raise ValueError(
+                    "The number of dimensions of data and number of axes of wcs donot match.")
+        super(NDCube, self).__init__(data, uncertainty=uncertainty, mask=mask,
+                                     wcs=wcs, meta=meta, unit=unit, copy=copy, **kwargs)
+
+    def pixel_to_world(self, origin=0):
```

Comment:
This function should allow the user to define the pixels of interest.  See API of https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L818